app/models/GuaranteeReference.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.RoutingOption.Gb
+import models.RoutingOption.Xi
+
+case class GuaranteeReference(value: String) extends AnyVal {
+  def countryCode: String =
+    value.drop(2).take(2)
+
+  def getRoutingOption: RoutingOption =
+    if (countryCode == Xi.prefix) Xi else Gb
```

Comment:
Did we get the confirmation that there are two guarantee services, XI and GB?

Reply:
I was told that
> THey will need routing to GB or XI depending on the country code in the GRN